src/internal/observable/of.ts

```diff
@@ -43,6 +41,9 @@ export function of<T, T2, T3, T4, T5, T6, T7, T8>(a: T, b: T2, c: T3, d: T4, e:
 export function of<T, T2, T3, T4, T5, T6, T7, T8, T9>(a: T, b: T2, c: T3, d: T4, e: T5, f: T6, g: T7, h: T8, i: T9):
   Observable<T | T2 | T3 | T4 | T5 | T6 | T7 | T8 | T9>;
 export function of<T>(...args: T[]): Observable<T>;
+
+/** @deprecated use {@link scheduled} instead `scheduled([a, b, c], scheduler)` */
```

Comment:
Actually, I've had another look at this and I don't think the signature can be moved. If it is, usage of a scheduler will match the signature above it and `T` will be inferred to be a union that includes the scheduler's type.

I think the signature will have to be left in its original location and that the deprecation for this particular signature should be removed instead.

Reply:
@cartant how about using Typescript Exclude operator in non-deprecated overloads? i.e. line 43 would look like this:

`export function of<T extends Exclude<any, SchedulerLike>>(...args: T[]): Observable<T>;`

Reply:
You are right, seems like Negated types are coming in Typescript 3.5.